libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -426,3 +437,52 @@ queryStakePoolDefaultVote ::
 queryStakePoolDefaultVote nes poolId =
   defaultStakePoolVote poolId (nes ^. nesEsL . epochStateStakePoolsL) $
     nes ^. nesEsL . esLStateL . lsCertStateL . certDStateL . accountsL
+
+-- | Used only for the `queryPoolState` query. This resembles the older way of
+-- representing StakePoolState in Ledger.
+data PState' = PState'
+  { psStakePoolParams :: !(Map (KeyHash 'StakePool) PoolParams)
+  , psFutureStakePoolParams :: !(Map (KeyHash 'StakePool) PoolParams)
+  , psRetiring' :: !(Map (KeyHash 'StakePool) EpochNo)
+  , psDeposits :: !(Map (KeyHash 'StakePool) Coin)
+  }
+  deriving (Show, Eq)
+
+instance EncCBOR PState' where
+  encCBOR (PState' a b c d) =
+    encodeListLen 4 <> encCBOR a <> encCBOR b <> encCBOR c <> encCBOR d
+
+instance DecShareCBOR PState' where
+  type Share PState' = Interns (KeyHash 'StakePool)
+  decSharePlusCBOR = decodeRecordNamedT "PState'" (const 4) $ do
+    psStakePoolParams <- decSharePlusLensCBOR (toMemptyLens _1 id)
+    psFutureStakePoolParams <- decSharePlusLensCBOR (toMemptyLens _1 id)
+    psRetiring' <- decSharePlusLensCBOR (toMemptyLens _1 id)
+    psDeposits <- decSharePlusLensCBOR (toMemptyLens _1 id)
+    pure PState' {psStakePoolParams, psFutureStakePoolParams, psRetiring', psDeposits}
```

Comment:
Results of queries do not need a `DecShareCBOR` instance, since sharing is only relevant for when reading large ledger state from a file.
```suggestion
instance DecCBOR PState' where
  decCBOR = decodeRecordNamed "PState'" (const 4) $ do
    psStakePoolParams <- decCBOR
    psFutureStakePoolParams <- decCBOR
    psRetiring' <- decCBOR
    psDeposits <- decCBOR
    pure PState' {psStakePoolParams, psFutureStakePoolParams, psRetiring', psDeposits}
```